edu.cuny.citytech.refactoring.common.core/src/edu/cuny/citytech/refactoring/common/core/ProcessorBasedRefactoring.java

```diff
@@ -0,0 +1,11 @@
+package edu.cuny.citytech.refactoring.common.core;
+
+import edu.cuny.citytech.refactoring.common.core.refactorings.RefactoringProcessor;
+
+public class ProcessorBasedRefactoring extends org.eclipse.ltk.core.refactoring.participants.ProcessorBasedRefactoring {
```

Comment:
Why do we need this class again?

Reply:
No. I used this to call `checkAllConditions`. Actually, I can call the method without this class.